doc/admin/pool_operations.md

```diff
@@ -177,13 +177,34 @@ with one ACE per line.
 
 ### Modifying a pool's ACL
 
+For all of these commands using an ACL file, the ACL file must be in the format
+noted above for pool creation.
+
+#### Overwriting the ACL
+
 To replace a pool's ACL with a new ACL:
 
 ```
 $ dmg pool overwrite-acl --pool <UUID> --acl-file <path>
 ```
 
-The ACL file must be in the format noted above for pool creation.
+#### Updating entries in an existing ACL
```

Comment:
The DAOS Tools proposal https://wiki.hpdd.intel.com/display/DC/DAOS+Tools had sketched out get-ac and set-ac and del-ac. get-acl gives the first one. Was set-ac(set-acl) considered? Does set-ac(set-acl) make sense for this, and possibly consolidating the overwrite case too? For example
set-acl --acl-file             (add/update entries in file in pool's ACL)
set-acl --entry                (add/update entry specified in pool's ACL)
set-acl --overwrite --acl-file (overwrite entire pool ACL)
or maybe even avoiding need for an --overwrite option have set-acl when given an --acl-file argument perform an overwrite of the pool's entire ACL?

Regarding possible consolidation at the code level: currently in the management service ds_mgmt_pool_update_acl() and  ds_mgmt_pool_overwrite_acl() these functions interact with pool service differently (via struct daos_acl versus daos_prop_t), that could maybe be unified around using struct daos_acl?

Reply:
We had a discussion about the tool during one of the control plane working group meetings, and settled on this set of external commands. The "set-acl --overwrite" idea was one of the options. There was a preference for having them be separate commands. It seemed a bit confusing to have a single command with a lot of different options attached to it. We thought of pulling the delete into it as well.

The underlying logic could probably be consolidated to an extent, at least sharing the boilerplate stuff, but it would require passing a boolean around, which implies to me that they are really separate functions. I liked having a generic CaRT set_prop/get_prop (was planning to change the get_acl CaRT call to a more generic get_prop), but of course that doesn't work for this case, where we need to do the read-modify-write inside the lock.

Reply:
On the tools it seems the conclusion is that the proposal text should reflect the updated choices that agree with the current/planned implementation.

On the underlying logic, what is the envisioned end state once all commands are implemented? Obviously the update will need to use acl as you described. I guess overwrite is already done through prop. get may (will?) be reimplemented in a more generic get prop framework.

Reply:
Updated the wiki page to match the current implementation.

Delete will also have to perform a specialized action to remove entries by principal, so will need its own CaRT command. It will accept an int (type) and string (name) indicating which principal to remove from the ACL. The other commands are intended to be as you described. I intend to do the work to change the get_acl to get_prop in the near future.